### PR TITLE
Go Peer Updates

### PR DESCRIPTION
**Pull request by TheDiscordian**
#23 

Complete:

- [x] TCP / WebSocket / QUIC (v0) support
- [x] AddrFactory addrs (for public addr specifying)
- [x] Bootstrap peer(s)

~Will tackle other two items tomorrow after sync.~

### CODE REVIEW

File: go-peer/main.go [hunk 1]
```diff
@@ -144,13 +156,45 @@ func main() {
...
+
+	opts = append(opts,
 		libp2p.Identity(privk),
 		libp2p.Transport(quicTransport.NewTransport),
+		libp2p.Transport(tcpTransport.NewTCPTransport),
 		libp2p.Transport(webtransport.New),
-		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/0/quic-v1", "/ip4/0.0.0.0/udp/0/quic-v1/webtransport", "/ip6/::/udp/0/quic-v1", "/ip6/::/udp/0/quic-v1/webtransport"),
+		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/0/quic-v1", "/ip4/0.0.0.0/udp/0/quic-v1/webtransport", "/ip6/::/udp/0/quic-v1", "/ip6/::/udp/0/quic-v1/webtransport",
+			"/ip4/0.0.0.0/udp/0/quic", "/ip6/::/udp/0/quic",
+			"/ip4/0.0.0.0/tcp/0", "/ip6/::/tcp/0",
```
Comment (p-shahi) [inline go-peer/main.go:192]:
```suggestion
		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9091/quic-v1", "/ip4/0.0.0.0/udp/9092/quic-v1/webtransport", "/ip4/0.0.0.0/tcp/0"),
```
we don't need to support IPv6 or QUIC v0 I think
Also changed made it run on port 90

Comment (marten-seemann) [reply go-peer/main.go:192]: Agreed regarding QUIC draft versions (it's not called v0). Why not IPv6 though?

Comment (p-shahi) [reply go-peer/main.go:192]: > Why not IPv6 though?

The other peers aren't listening on v6 so might as well disable it here

**Review by p-shahi — APPROVED** [review]

File: go-peer/main.go (hunk 1 above)
Comment (TheDiscordian) [reply go-peer/main.go]: FWIW ipv6 is usually easier to peer with (no NAT), but I'm neutral otherwise if the other nodes aren't going to support it

Comment (marten-seemann) [reply go-peer/main.go]: Not that it matters much here, but I'd be a bit sad building something in 2023 that doesn't do IPv6.

**Pull request merged**